{"matches":[{"vulnerability":{"id":"GHSA-248v-346w-9cwc","dataSource":"https://github.com/advisories/GHSA-248v-346w-9cwc","namespace":"github:language:python","severity":"Low","urls":[],"description":"Certifi removes GLOBALTRUST root certificate","cvss":[],"epss":[{"cve":"CVE-2024-39689","epss":0.01928,"percentile":0.82507,"date":"2025-06-03"}],"fix":{"versions":["2024.7.4"],"state":"fixed"},"advisories":[],"risk":0.5783999999999999},"relatedVulnerabilities":[{"id":"CVE-2024-39689","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-39689","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/certifi/python-certifi/commit/bd8153872e9c6fc98f4023df9c2deaffea2fa463","https://github.com/certifi/python-certifi/security/advisories/GHSA-248v-346w-9cwc","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/XpknYMPO8dI","https://github.com/certifi/python-certifi/commit/bd8153872e9c6fc98f4023df9c2deaffea2fa463","https://github.com/certifi/python-certifi/security/advisories/GHSA-248v-346w-9cwc","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/XpknYMPO8dI","https://security.netapp.com/advisory/ntap-20241206-0001/"],"description":"Certifi is a curated collection of Root Certificates for validating the trustworthiness of SSL certificates while verifying the identity of TLS hosts. Certifi starting in 2021.5.30 and prior to 2024.7.4 recognized root certificates from `GLOBALTRUST`. Certifi 2024.7.04 removes root certificates from `GLOBALTRUST` from the root store. These are in the process of being removed from Mozilla's trust store. `GLOBALTRUST`'s root certificates are being removed pursuant to an investigation which identified \"long-running and unresolved compliance issues.\"","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-39689","epss":0.01928,"percentile":0.82507,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"certifi","version":"2023.7.22"}},"found":{"versionConstraint":">=2021.5.30,<2024.7.4 (python)","vulnerabilityID":"GHSA-248v-346w-9cwc"},"fix":{"suggestedVersion":"2024.7.4"}}],"artifact":{"id":"646b835b464af341","name":"certifi","version":"2023.7.22","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/certifi-2023.7.22.dist-info/METADATA","accessPath":"/usr/lib/python3/dist-packages/certifi-2023.7.22.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/certifi-2023.7.22.dist-info/RECORD","accessPath":"/usr/lib/python3/dist-packages/certifi-2023.7.22.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/usr/lib/python3/dist-packages/certifi-2023.7.22.dist-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/certifi-2023.7.22.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MPL-2.0"],"cpes":["cpe:2.3:a:certifi:certifi:2023.7.22:*:*:*:*:python:*:*"],"purl":"pkg:pypi/certifi@2023.7.22","upstreams":[]}},{"vulnerability":{"id":"GHSA-jjg7-2v4v-x38h","dataSource":"https://github.com/advisories/GHSA-jjg7-2v4v-x38h","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Internationalized Domain Names in Applications (IDNA) vulnerable to denial of service from specially crafted inputs to idna.encode","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":6.2,"exploitabilityScore":2.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3651","epss":0.00436,"percentile":0.62113,"date":"2025-06-03"}],"fix":{"versions":["3.7"],"state":"fixed"},"advisories":[],"risk":0.24416000000000004},"relatedVulnerabilities":[{"id":"CVE-2024-3651","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-3651","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/kjd/idna/commit/1d365e17e10d72d0b7876316fc7b9ca0eebdd38d","https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb","https://github.com/kjd/idna/commit/1d365e17e10d72d0b7876316fc7b9ca0eebdd38d","https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb"],"description":"A vulnerability was identified in the kjd/idna library, specifically within the `idna.encode()` function, affecting version 3.6. The issue arises from the function's handling of crafted input strings, which can lead to quadratic complexity and consequently, a denial of service condition. This vulnerability is triggered by a crafted input that causes the `idna.encode()` function to process the input with considerable computational load, significantly increasing the processing time in a quadratic manner relative to the input size.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":6.2,"exploitabilityScore":2.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3651","epss":0.00436,"percentile":0.62113,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"idna","version":"3.4"}},"found":{"versionConstraint":"<3.7 (python)","vulnerabilityID":"GHSA-jjg7-2v4v-x38h"},"fix":{"suggestedVersion":"3.7"}}],"artifact":{"id":"02efbdad08258065","name":"idna","version":"3.4","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/idna-3.4.dist-info/METADATA","accessPath":"/usr/lib/python3/dist-packages/idna-3.4.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/idna-3.4.dist-info/RECORD","accessPath":"/usr/lib/python3/dist-packages/idna-3.4.dist-info/RECORD","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:kim_davies_<kim_project:python-idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim_project:python_idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kimproject:python-idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kimproject:python_idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim_project:idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim-davies-<kim:python-idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim-davies-<kim:python_idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim:python-idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim:python_idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kimproject:idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:python-idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:python_idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:python-idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:python_idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim-davies-<kim:idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim:idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python-idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python_idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:idna:python-idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:idna:python_idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:python:idna:3.4:*:*:*:*:*:*:*","cpe:2.3:a:idna:idna:3.4:*:*:*:*:*:*:*"],"purl":"pkg:pypi/idna@3.4","upstreams":[]}},{"vulnerability":{"id":"GHSA-g4mx-q9vg-27p4","dataSource":"https://github.com/advisories/GHSA-g4mx-q9vg-27p4","namespace":"github:language:python","severity":"Medium","urls":[],"description":"urllib3's request body not stripped after redirect from 303 status changes request method to GET","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-45803","epss":0.00062,"percentile":0.1968,"date":"2025-06-03"}],"fix":{"versions":["2.0.7"],"state":"fixed"},"advisories":[],"risk":0.02852},"relatedVulnerabilities":[{"id":"CVE-2023-45803","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-45803","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9","https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/","https://www.rfc-editor.org/rfc/rfc9110.html#name-get","https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9","https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/","https://www.rfc-editor.org/rfc/rfc9110.html#name-get"],"description":"urllib3 is a user-friendly HTTP client library for Python. urllib3 previously wouldn't remove the HTTP request body when an HTTP redirect response using status 301, 302, or 303 after the request had its method changed from one that could accept a request body (like `POST`) to `GET` as is required by HTTP RFCs. Although this behavior is not specified in the section for redirects, it can be inferred by piecing together information from different sections and we have observed the behavior in other major HTTP client implementations like curl and web browsers. Because the vulnerability requires a previously trusted service to become compromised in order to have an impact on confidentiality we believe the exploitability of this vulnerability is low. Additionally, many users aren't putting sensitive data in HTTP request bodies, if this is the case then this vulnerability isn't exploitable. Both of the following conditions must be true to be affected by this vulnerability: 1. Using urllib3 and submitting sensitive information in the HTTP request body (such as form data or JSON) and 2. The origin service is compromised and starts redirecting using 301, 302, or 303 to a malicious peer or the redirected-to service becomes compromised. This issue has been addressed in versions 1.26.18 and 2.0.7 and users are advised to update to resolve this issue. Users unable to update should disable redirects for services that aren't expecting to respond with redirects with `redirects=False` and disable automatic redirects with `redirects=False` and handle 301, 302, and 303 redirects manually by stripping the HTTP request body.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-45803","epss":0.00062,"percentile":0.1968,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"urllib3","version":"2.0.6"}},"found":{"versionConstraint":">=2.0.0,<2.0.7 (python)","vulnerabilityID":"GHSA-g4mx-q9vg-27p4"},"fix":{"suggestedVersion":"2.0.7"}}],"artifact":{"id":"38f9aedb76c83b5f","name":"urllib3","version":"2.0.6","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/urllib3-2.0.6.dist-info/METADATA","accessPath":"/usr/lib/python3/dist-packages/urllib3-2.0.6.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/urllib3-2.0.6.dist-info/RECORD","accessPath":"/usr/lib/python3/dist-packages/urllib3-2.0.6.dist-info/RECORD","annotations":{"evidence":"supporting"}}],"language":"python","licenses":[],"cpes":["cpe:2.3:a:python:urllib3:2.0.6:*:*:*:*:*:*:*"],"purl":"pkg:pypi/urllib3@2.0.6","upstreams":[]}},{"vulnerability":{"id":"GHSA-34jh-p97f-mpxf","dataSource":"https://github.com/advisories/GHSA-34jh-p97f-mpxf","namespace":"github:language:python","severity":"Medium","urls":[],"description":"urllib3's Proxy-Authorization request header isn't stripped during cross-origin redirects","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.4,"exploitabilityScore":0.8,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-37891","epss":0.00033,"percentile":0.08049,"date":"2025-06-03"}],"fix":{"versions":["2.2.2"],"state":"fixed"},"advisories":[],"risk":0.01551},"relatedVulnerabilities":[{"id":"CVE-2024-37891","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-37891","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e","https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf","https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e","https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf","https://security.netapp.com/advisory/ntap-20240822-0003/","https://www.vicarius.io/vsociety/posts/proxy-authorization-header-handling-vulnerability-in-urllib3-cve-2024-37891"],"description":"urllib3 is a user-friendly HTTP client library for Python. When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected. However, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects. Because this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident. Users should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach. We believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited: 1. Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support. 2. Not disabling HTTP redirects. 3. Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin. Users are advised to update to either version 1.26.19 or version 2.2.2. Users unable to upgrade may use the `Proxy-Authorization` header with urllib3's `ProxyManager`, disable HTTP redirects using `redirects=False` when sending requests, or not user the `Proxy-Authorization` header as mitigations.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.4,"exploitabilityScore":0.8,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-37891","epss":0.00033,"percentile":0.08049,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"urllib3","version":"2.0.6"}},"found":{"versionConstraint":">=2.0.0,<2.2.2 (python)","vulnerabilityID":"GHSA-34jh-p97f-mpxf"},"fix":{"suggestedVersion":"2.2.2"}}],"artifact":{"id":"38f9aedb76c83b5f","name":"urllib3","version":"2.0.6","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/urllib3-2.0.6.dist-info/METADATA","accessPath":"/usr/lib/python3/dist-packages/urllib3-2.0.6.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/urllib3-2.0.6.dist-info/RECORD","accessPath":"/usr/lib/python3/dist-packages/urllib3-2.0.6.dist-info/RECORD","annotations":{"evidence":"supporting"}}],"language":"python","licenses":[],"cpes":["cpe:2.3:a:python:urllib3:2.0.6:*:*:*:*:*:*:*"],"purl":"pkg:pypi/urllib3@2.0.6","upstreams":[]}},{"vulnerability":{"id":"GHSA-9wx4-h78v-vm56","dataSource":"https://github.com/advisories/GHSA-9wx4-h78v-vm56","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Requests `Session` object does not verify requests after making first request with verify=False","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.6,"exploitabilityScore":0.4,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-35195","epss":0.00009,"percentile":0.00549,"date":"2025-06-03"}],"fix":{"versions":["2.32.0"],"state":"fixed"},"advisories":[],"risk":0.004770000000000001},"relatedVulnerabilities":[{"id":"CVE-2024-35195","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-35195","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac","https://github.com/psf/requests/pull/6655","https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IYLSNK5TL46Q6XPRVMHVWS63MVJQOK4Q/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/N7WP6EYDSUOCOJYHDK5NX43PYZ4SNHGZ/","https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac","https://github.com/psf/requests/pull/6655","https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IYLSNK5TL46Q6XPRVMHVWS63MVJQOK4Q/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/N7WP6EYDSUOCOJYHDK5NX43PYZ4SNHGZ/"],"description":"Requests is a HTTP library. Prior to 2.32.0, when making requests through a Requests `Session`, if the first request is made with `verify=False` to disable cert verification, all subsequent requests to the same host will continue to ignore cert verification regardless of changes to the value of `verify`. This behavior will continue for the lifecycle of the connection in the connection pool. This vulnerability is fixed in 2.32.0.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.6,"exploitabilityScore":0.4,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-35195","epss":0.00009,"percentile":0.00549,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"requests","version":"2.31.0"}},"found":{"versionConstraint":"<2.32.0 (python)","vulnerabilityID":"GHSA-9wx4-h78v-vm56"},"fix":{"suggestedVersion":"2.32.0"}}],"artifact":{"id":"ca0519ae09f1d9e3","name":"requests","version":"2.31.0","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/requests-2.31.0.dist-info/METADATA","accessPath":"/usr/lib/python3/dist-packages/requests-2.31.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/requests-2.31.0.dist-info/RECORD","accessPath":"/usr/lib/python3/dist-packages/requests-2.31.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/usr/lib/python3/dist-packages/requests-2.31.0.dist-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/requests-2.31.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache 2.0"],"cpes":["cpe:2.3:a:python:requests:2.31.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/requests@2.31.0","upstreams":[]}}],"source":{"type":"sbom-file","target":"wordbook-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/wordbook-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/wordbook-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-04T01:32:23Z_1749035557.tar.zst?checksum=sha256%3A77238f4ffbc7cf72bf4dedabbb21a3d0dd3ade90fb5d9df7c4da2870f8d1a9b7","built":"2025-06-04T11:12:37Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-04T01:32:23Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-04T01:32:32Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-04T01:32:33Z","input":"xxh64:03c9c4d182d77791"},"debian":{"captured":"2025-06-04T01:32:49Z","input":"xxh64:169ceb8fbcb8d6f1"},"epss":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:bf87c4c234d02346"},"github":{"captured":"2025-06-04T01:32:30Z","input":"xxh64:3dab0be50635e057"},"kev":{"captured":"2025-06-04T01:32:45Z","input":"xxh64:a934e75f468f4e09"},"mariner":{"captured":"2025-06-04T01:32:43Z","input":"xxh64:ac698752085c186f"},"nvd":{"captured":"2025-06-04T01:33:28Z","input":"xxh64:389887843a59d29c"},"oracle":{"captured":"2025-06-04T01:32:35Z","input":"xxh64:0b8aabcfb6505b62"},"rhel":{"captured":"2025-06-04T04:49:10Z","input":"xxh64:947d0caeb19e9ccc"},"sles":{"captured":"2025-06-04T01:32:34Z","input":"xxh64:1c50b1219296628a"},"ubuntu":{"captured":"2025-06-04T01:33:14Z","input":"xxh64:71e22d5c5bbe7d14"},"wolfi":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:e68fca5a870c6769"}}},"timestamp":"2025-06-04T19:51:51.30121+01:00"}}
